R/paramatch.r

```diff
@@ -1,49 +1,73 @@
-#' Count the occurances of the N most repeated words in a paragraph
+#' Mungebit to count the occurances of the N most repeated words in a dataframe column.
 #'
-#' Paragraphs of M lines for N words will return a data frame of N + 1 columns and M rows each counting the occurance in the particular line.
+#' Trains to find the most repeated words.  Then counts the number of matches for dataframes passed to it.
 #'
 #' This conducts full word matching only.  Substring matches do not count towards the total.
 #'
-#' @param input an atomic vector. The paragraph of text to analyse.
+#' @param dataframe
+#' @param col character. The column name pertaining to paragraphs for matching.
 #' @param top_n_words integer.  The number of top strings to count in each line.
 #' @param suppress.input logical.  Suppresses the Input column of the output data frame.
+#' @param blacklist character. Prevents undesired words participating int he output.
 #' @return numeric. Number of occurances of the particular word in each line.
-#' @export
+#' @author Mike Bruno
 #' @examples
-#' paragraph <- c("This is an example.", "Great code uses examples like this example.")
-#' paramatch(paragraph, 2)
-paramatch <- function(input, top_n_words = 5, suppress.input = FALSE) {
+#' paragraph <- data.frame(id = c(1:2), text = c("This is an example.", "Great code uses examples like this example."), stringsAsFactors = FALSE)
+#' paramatch(paragraph, "text", 2)
+
+#' @export
+paramatch <- function(dataframe, col, top_n_words = 5, suppress.input = FALSE, blacklist = c()) {
+  # Grab the string vector
+  paragraph_col <- dataframe[[col]]
   # Make sure inputs are valid
-    stopifnot(is.character(input))
-    stopifnot(is.numeric(top_n_words))
-    stopifnot(top_n_words >= 1)
-    stopifnot(top_n_words %% 1 == 0)
-    stopifnot(is.logical(suppress.input))
+  stopifnot(is.character(paragraph_col))
+  stopifnot(is.numeric(top_n_words))
+  stopifnot(top_n_words >= 1)
+  stopifnot(top_n_words %% 1 == 0)
+  stopifnot(is.logical(suppress.input))
   # Standardize the input
-    paragraph <- toupper(input)
-    paragraph <- gsub("[[:punct:]]", " ", paragraph)
-    paragraph <- gsub("[[:space:]]+", " ", paragraph)
```

Comment:
If you wanna get really fancy you can try to find a package to reduce various cases/tenses to a single common word stem, e.g. the tm package: http://cran.r-project.org/web/packages/tm/tm.pdf